src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -199,9 +207,20 @@ private async Task EnqueueProcessSnapshotWorkerAsync(Document document, Cancella
                     return;
                 }
 
-                // preemptively parse file in background so that when we are called from tagger from UI thread, we have tree ready.
-                // F#/typescript and other languages that doesn't support syntax tree will return null here.
-                _ = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                // TODO - Cleanup once experiment completed - https://github.com/dotnet/roslyn/projects/45#card-27261853
+                if (ShouldLogLocalTelemetry(document.Project.Language))
+                {
+                    using (new RequestLatencyTracker(SyntacticLspLogger.RequestType.SyntacticTagger))
+                    {
+                        _ = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Rather than duplicating the code consider doing something like:
```csharp
var latencyTracker = ShouldLogLocalTelemetry(document.Project.Language) ? new … : null;
using (latencyTracker)
{
}
```

That way we will conditionally dispose.